wallet-new/docs/spec.tex

```diff
@@ -3009,38 +3025,533 @@ \subsection{Use cases}
 ability to `empty their wallet' (i.e., create a single transaction that
 spends all of the wallet's UTxO, sending it to a single output address.)
 
-\subsection{Implementation notes}
+\subsection{Self organisation}
+\label{sec:selforganisation}
+
+The term `self organization' refers to the emergence of complex behaviour
+(typically in biological systems) from simple rules and random fluctuations.
+In this section we will see how we can take advantage of self organization
+to design a simple yet effective coin selection algorithm.
+
+In this section we present the conclusions of an in-depth study of coin
+selection that we have done as part of the the redesign of the Cardano wallet.
+We will describe some of the problems and trade-offs that need to be made. We
+then propose a novel coin selection algorithm. The algorithm is efficient and
+straight-forward to implement, and through studing simulations of the algorithm
+under various conditions we will see that it is very effective.
+As our starting point, we will use the random algorithm described by Mark
+Erhardt in his masters thesis \citep{Erhardt:thesis}.
+
+\subsection{Dust}
+
+An obvious strategy that many coin selection algorithms use in some form or
+other is ``try to get as close to the requested value as possible''. The problem
+with such an approach is that it tends to create a lot of \emph{dust}: small
+unspent outputs that remain unused in the user's wallet because they're not
+particularly useful. For example, consider the `largest first' algorithm: a
+simple algorithm which considers all unspent outputs of the wallet in order of
+size, adding them to a running total until it has covered the requested amount.
+Figure~\ref{fig:inputselection:normal_3to1_largest} shows the effect of this
+algorithm.
+
+\begin{figure}[p]
+\begin{center}
+\scriptsize
+\begin{tabular}{ll}
+\input{spec/normal-3to1-largest.tex} &
+\input{spec/normal-3to1-largest-distr.tex}
+\end{tabular}
+\end{center}
+\caption{\label{fig:inputselection:normal_3to1_largest}
+  Simulation of largest-first coin selection. Main histogram shows UTxO entries;
+  inset graph shows UTxO balance in blue and UTxO size in red, histogram
+  top-right shows number of inputs per transaction, graph bottom right shows the
+  change:payment ratio (more on that below). Graph on the far right shows the
+  distribution of deposits (blue, left axis) versus payments (red, right axis).
+  In this case, both are normally distributed with a mean of 1000 and 3000
```

Comment:
Probably before we start talking about normally distributed, we should explain the simulation uses a stream of events of deposits and withdrawals with assumed distributions. Basically we should explain our simulation strategy at some point.